src/equations/reference_data.jl

```diff
@@ -52,7 +52,7 @@ for Cartesian and covariant formulations.
 @inline function initial_condition_gaussian(x, t)
     RealT = eltype(x)
 
-    a = EARTH_RADIUS  # radius of the sphere in metres
+    a = sqrt(x[1]^2 + x[2]^2 + x[3]^2) #radius of the sphere
```

Comment:
This is needed when the geometry is approximated with polynomials and not "exactly". In such cases, we can get issues when computing the latitude `asin(x[3] / a)`, as `x[3]` might be a bit larger than `a`.